Source/ChocolateyGui.Common.Windows/Utilities/ChocolateyMessageBox.cs

```diff
@@ -0,0 +1,137 @@
+            dummyWindow.Show();
+            var result = MessageBox.Show(messageBoxText, caption, button, icon);
+            dummyWindow.Show();
+            return result;
+        }
+
+        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult, MessageBoxOptions options)
+        {
+            var dummyWindow = DummyWindow();
+            dummyWindow.Show();
+            var result = MessageBox.Show(owner, messageBoxText, caption, button, icon, defaultResult, options);
+            dummyWindow.Show();
+            return result;
+        }
+
+        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult)
+        {
+            var dummyWindow = DummyWindow();
+            dummyWindow.Show();
+            var result = MessageBox.Show(owner, messageBoxText, caption, button, icon, defaultResult);
+            dummyWindow.Show();
+            return result;
+        }
+
+        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon)
+        {
+            var dummyWindow = DummyWindow();
+            dummyWindow.Show();
+            var result = MessageBox.Show(owner, messageBoxText, caption, button, icon);
+            dummyWindow.Show();
+            return result;
+        }
+
+        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button)
+        {
+            var dummyWindow = DummyWindow();
+            dummyWindow.Show();
+            var result = MessageBox.Show(owner, messageBoxText, caption, button);
+            dummyWindow.Show();
+            return result;
+        }
+
+        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption)
+        {
+            var dummyWindow = DummyWindow();
+            dummyWindow.Show();
+            var result = MessageBox.Show(owner, messageBoxText, caption);
+            dummyWindow.Show();
+            return result;
+        }
+
+        public static MessageBoxResult Show(Window owner, string messageBoxText)
+        {
+            var dummyWindow = DummyWindow();
+            dummyWindow.Show();
+            var result = MessageBox.Show(owner, messageBoxText);
+            dummyWindow.Show();
+            return result;
+        }
+
+        private static Window DummyWindow()
+        {
+            return new Window
+            {
+                AllowsTransparency = true,
+                Background = System.Windows.Media.Brushes.Transparent,
+                WindowStyle = WindowStyle.None,
+                Top = 0,
+                Left = 0,
+                Width = 1,
+                Height = 1,
+                ShowInTaskbar = false
+            };
+        }

```

Comment:
There is quite a bit of duplication happening here.  This can be prevented by calling the overloaded version of each constructor from the previous one.  Something like the following:

```suggestion
public static MessageBoxResult Show(string messageBoxText)
        {
            return Show(messageBoxText, string.Empty);
        }

        public static MessageBoxResult Show(string messageBoxText, string caption)
        {
            return Show(messageBoxText, caption, MessageBoxButton.OK);
        }

        public static MessageBoxResult Show(string messageBoxText, string caption, MessageBoxButton button)
        {
            return Show(messageBoxText, caption, button, MessageBoxImage.None);
        }

        public static MessageBoxResult Show(string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon)
        {
            return Show(messageBoxText, caption, button, icon, MessageBoxResult.OK);
        }

        public static MessageBoxResult Show(string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult)
        {
            return Show(messageBoxText, caption, button, icon, defaultResult, MessageBoxOptions.None);
        }

        public static MessageBoxResult Show(string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult, MessageBoxOptions options)
        {
            var dummyWindow = DummyWindow();
            dummyWindow.Show();
            var result = MessageBox.Show(messageBoxText, caption, button, icon, defaultResult, options);
            dummyWindow.Show();
            return result;
        }

        public static MessageBoxResult Show(Window owner, string messageBoxText)
        {
            return Show(owner, messageBoxText, string.Empty);
        }

        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption)
        {
            return Show(owner, messageBoxText, caption, MessageBoxButton.OK);
        }

        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button)
        {
            return Show(owner, messageBoxText, caption, button, MessageBoxImage.None);
        }

        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon)
        {
            return Show(owner, messageBoxText, caption, button, icon, MessageBoxResult.OK);
        }

        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult)
        {
            return Show(owner, messageBoxText, caption, button, icon, defaultResult, MessageBoxOptions.None);
        }

        public static MessageBoxResult Show(Window owner, string messageBoxText, string caption, MessageBoxButton button, MessageBoxImage icon, MessageBoxResult defaultResult, MessageBoxOptions options)
        {
            var dummyWindow = DummyWindow();
            dummyWindow.Show();
            var result = MessageBox.Show(owner, messageBoxText, caption, button, icon, defaultResult, options);
            dummyWindow.Show();
            return result;
        }

        private static Window DummyWindow()
        {
            return new Window
            {
                AllowsTransparency = true,
                Background = System.Windows.Media.Brushes.Transparent,
                WindowStyle = WindowStyle.None,
                Top = 0,
                Left = 0,
                Width = 1,
                Height = 1,
                ShowInTaskbar = false
            };
        }
```